src/components/userFeedback/UserFeedback.spec.tsx

```diff
@@ -0,0 +1,143 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+import { UserFeedback, UserFeedbackState, IUserFeedbackProps, TextColorClasses } from './UserFeedback';
+
+// Until Webpack provided plugins works with TS 2.0
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+
+describe('<UserFeedback>', () => {
+  const defaultTextColorClass: string = TextColorClasses.default;
+  const errorTextColorClass: string = TextColorClasses.error;
+
+  const generateProps = (feedbackText: string, state: string, extraClasses?: string, displayOnShow?: string): IUserFeedbackProps => {
+    return displayOnShow ? { feedbackText, state, extraClasses, displayOnShow } : { feedbackText, state, extraClasses };
+  };
+
+  const defaultProps: IUserFeedbackProps = generateProps('', '', '');
+  const defaultPropsWithDisplay: IUserFeedbackProps = generateProps('', '', '', 'block');
+
+  const getShallowOutput = (props: IUserFeedbackProps): ShallowWrapper<IUserFeedbackProps, any> => {
+    return shallow(<UserFeedback {...props} />);
+  };
+
+  describe('"Shallow rendered" component', () => {
+    describe('rendering', () => {
+      it('should render without errors', () => {
+        expect(() => getShallowOutput(defaultProps)).not.toThrow();
+        expect(() => getShallowOutput(defaultPropsWithDisplay)).not.toThrow();
+        expect(() => getShallowOutput(generateProps('', '', ''))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.VALID, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.WARNING, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.ERROR, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.VALID, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.WARNING, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.ERROR, 'm1', 'block'))).not.toThrow();
+      });
+    });
+
+    describe('text content', () => {
+      it('should contain the exact text passed to prop feedbackText', () => {
+        let testText = 'testing';
+        let wrongText: string;
+
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).toBe(testText);
+
+        testText = 'hello this is a longer test string over here';
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).toBe(testText);
+
+
+        testText = 'hello this is a test over here';
+        wrongText = 'hello this is the wrong text over here';
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).not.toBe(wrongText);
```

Comment:
Thanks JP. The negative test is to make sure that the positive tests were not false positive, would that make sense?